easybuild/toolchains/compiler/llvm_compilers.py

```diff
@@ -0,0 +1,193 @@
+        # -funsafe-math-optimizations: allow unsafe math optimizations (implies -fassociative-math, -fno-signed-zeros,
+        # -freciprocal-math).
+        # -ffast-math: an umbrella flag that enables all optimizations listed above, provides preprocessor macro
+        # __FAST_MATH__.
+        #
+        # Using -fno-fast-math is equivalent to disabling all individual optimizations, see
+        # http://llvm.org/viewvc/llvm-project/cfe/trunk/lib/Driver/Tools.cpp?view=markup (lines 2100 and following)
+        #
+        # 'strict', 'precise' and 'defaultprec' are all ISO C++ and IEEE complaint, but we explicitly specify details
+        # flags for strict and precise for robustness against future changes.
+        'strict': ['-fno-fast-math'],
+        'precise': ['-fno-unsafe-math-optimizations'],
+        'defaultprec': [],
+        'loose': ['-ffast-math', '-fno-unsafe-math-optimizations'],
+        'veryloose': ['-ffast-math'],
+        'vectorize': {False: '-fno-vectorize', True: '-fvectorize'},
+        DEFAULT_OPT_LEVEL: ['-O2'],
+
+        'lld_undefined_version': ['-Wl,--undefined-version'],
+        'no_unused_args': ['-Wno-unused-command-line-argument'],
+        'no_int_conversion_error': ['-Wno-error=int-conversion'],
+    }
+
+    # Ensure that compiler options only appear once, so that arguments do not appear multiple times when compiling.
+    COMPILER_OPTIONS = list({
+        *(Compiler.COMPILER_OPTIONS or []),
+        'lld_undefined_version'
+    })
+
+    # Options only available for Clang compiler
+    COMPILER_C_UNIQUE_OPTIONS = list({
+        *(Compiler.COMPILER_C_UNIQUE_OPTIONS or []),
+        'no_unused_args',
+        'no_int_conversion_error'
+    })
+
+    # Options only available for Flang compiler
+    COMPILER_F_UNIQUE_OPTIONS = Compiler.COMPILER_F_UNIQUE_OPTIONS
+
+    # used when 'optarch' toolchain option is enabled (and --optarch is not specified)
+    COMPILER_OPTIMAL_ARCHITECTURE_OPTION = {
+        **(Compiler.COMPILER_OPTIMAL_ARCHITECTURE_OPTION or {}),
+        (systemtools.POWER, systemtools.POWER): '-mcpu=native',  # no support for march=native on POWER
+        (systemtools.POWER, systemtools.POWER_LE): '-mcpu=native',  # no support for march=native on POWER
+        (systemtools.X86_64, systemtools.AMD): '-march=native',
+        (systemtools.X86_64, systemtools.INTEL): '-march=native',
+    }
+
+    # used with --optarch=GENERIC
+    COMPILER_GENERIC_OPTION = {
+        **(Compiler.COMPILER_GENERIC_OPTION or {}),
+        (systemtools.RISCV64, systemtools.RISCV): '-march=rv64gc -mabi=lp64d',  # default for -mabi is system-dependent
+        (systemtools.X86_64, systemtools.AMD): '-march=x86-64 -mtune=generic',
+        (systemtools.X86_64, systemtools.INTEL): '-march=x86-64 -mtune=generic',
+    }
+
+    COMPILER_CC = 'clang'
+    COMPILER_CXX = 'clang++'
+
+    COMPILER_F77 = 'flang'
+    COMPILER_F90 = 'flang'
+    COMPILER_FC = 'flang'
+
+    LINKERS = ['lld', 'ld.lld', 'ld64.lld']
```

Comment:
let's make this new constant an immutable value:
```suggestion
    LINKERS = ('lld', 'ld.lld', 'ld64.lld')
```

That should also be done for other constants, but that needs a broader effort across existing supported compilers (that's out of scope for this PR)